### DISCUSSION TITLE
Dependency on package versioning script

### DISCUSSION THREAD

**Discussion by tompahoward** [vercel/turborepo#6333] [Help]
Hi,

I'm having trouble figuring out how to write the pipeline I'm after. 

First, I have a normal `build` task that depends on `^build`.

I also want to have a `publish` task. It depends on `build` and `//#version`, but `//#version` needs to complete before `build` is run. Or more specifically, the `build` task depends on `//#version`, but only when it's run as a dependency of the `publish` task.

How do I describe this in a my turbo pipeline?

Thanks in advance

**Comment by tompahoward** [comment]
I think I have a related example from https://turbo.build/repo/docs/core-concepts/monorepos/running-tasks#running-tasks-from-the-root

```json
{
  "$schema": "https://turbo.build/schema.json",
  "pipeline": {
    "build": {
      "dependsOn": ["^build"],
      "outputs": [".next/**", "!.next/cache/**", ".svelte-kit/**"]
    },
    "test": {
      "dependsOn": ["^build"],
    },
    // This will cause the "test" script to be included when
    // "turbo run test" is run
    "//#test": {
      "dependsOn": [],
    },
    // This will cause the "format" script in the root package.json
    // to be run when "turbo run format" is run. Since the general
    // "format" task is not defined, only the root's "format" script
    // will be run.
    "//#format": {
      "dependsOn": [],
      "outputs": ["dist/**/*"],
      "inputs": ["version.txt"]
    }
  }
}
```

In the above example, how would I get `//#test` to run after all the `test` tasks?

**Reply by tompahoward** [reply]
Anyone? 🙏

**Reply by anthonyshew** [reply]
Do you think you could make me a small reproduction with your best effort? I'm having a hard time grokking the dependency graph you're looking for. 😄
